swarm/network/simulation/simulation.go

```diff
@@ -76,6 +80,9 @@ func New(services map[string]ServiceFunc) (s *Simulation) {
 
 	adapterServices := make(map[string]adapters.ServiceFunc, len(services))
 	for name, serviceFunc := range services {
+		// Scope this variables correctly
+		// as they will be in the adapterServices[name] function accessed later.
+		name, serviceFunc := name, serviceFunc
```

Comment:
I suppose that we never started all services up until now, and newly created TestAddNodeMultipleServices was a missing test case from start.

Reply:
I totally disagree. Are you saying that we should change a common approach in go because you are not familiar with it? There is even a comment above. I am also interested what others think.

With different variable names, the original ones are still available and can be used instead the new ones which opens the possibility for the same issue that should be solved with scoping. With the same variable names there is no such risk.

Reply:
> Are you saying that we should change a common approach in go because you are not familiar with it?

No, that's not what I'm saying. In fact, I didn't even know it was a common approach in `go`. I'm sure I should have.

What I _am_ saying is I actually deleted code like this before because it looked nonsensical, and I'm pretty sure it even slipped through code review, which it shouldn't have if it's glaringly obvious. So let's make sure everyone in the team recognizes this trick. Maybe the comment is enough, I dunno.

Apologies for my ignorance.

Reply:
Thanks @nloash. If you have any suggestions how the comment can be improved, please provide it.

Reply:
I've posted in hq to raise awareness.

> the original ones are still available and can be used instead the new ones which opens the possibility for the same issue that should be solved with scoping. 

I understand the convenience, but I really don't like the style. I believe we should explicitly know wha

But then, who am I to contend with the perfect coder conditioning oracle that is Google :)

Reply:
Great, thanks. :)

Reply:
I think comment is fine for this notation. We should be aware of this construct and just raise awareness of it. I personally would not add a comment if I write code that needs to scope the loop vars.

https://github.com/golang/go/wiki/CommonMistakes